Sources/OTelCore/OTel+Configuration.swift

```diff
@@ -198,23 +465,158 @@ extension OTel.Configuration.LogsConfiguration {
+        /// Path to client certificate/chain trust for mTLS communication, in PEM format.
+        ///
+        /// Must be provided together with the client private key for mTLS to work.
+        ///
+        /// - Environment variable(s):
+        ///   - `OTEL_EXPORTER_OTLP_CLIENT_CERTIFICATE`
+        ///   - `OTEL_EXPORTER_OTLP_TRACES_CLIENT_CERTIFICATE`
+        ///   - `OTEL_EXPORTER_OTLP_METRICS_CLIENT_CERTIFICATE`
+        ///   - `OTEL_EXPORTER_OTLP_LOGS_CLIENT_CERTIFICATE`
+        /// - Default value: `nil`
+        /// - Notes: Signal-specific configuration takes precedence over the general configuration.
         public var clientCertificateFilePath: String?
+
+        /// Key-value pairs used as headers for gRPC or HTTP requests.
+        ///
+        /// Additional HTTP headers to include in export requests. Headers are specified
+        /// as key-value pairs and can be used for authentication, routing, or other purposes.
+        ///
+        /// - Environment variable(s):
+        ///   - `OTEL_EXPORTER_OTLP_HEADERS`
+        ///   - `OTEL_EXPORTER_OTLP_TRACES_HEADERS`
+        ///   - `OTEL_EXPORTER_OTLP_METRICS_HEADERS`
+        ///   - `OTEL_EXPORTER_OTLP_LOGS_HEADERS`
+        /// - Default value: Empty array
+        /// - Format: W3C Baggage format: `key1=value1,key2=value2`
+        /// - Notes: Signal-specific configuration takes precedence over the general configuration.
         public var headers: [(String, String)]
+
+        /// Compression method for request payload.
+        ///
+        /// Controls whether and how telemetry data is compressed before transmission to reduce
+        /// network bandwidth usage. Supported compression algorithms vary by transport protocol.
+        ///
+        /// - Environment variable(s):
+        ///   - `OTEL_EXPORTER_OTLP_COMPRESSION`
+        ///   - `OTEL_EXPORTER_OTLP_TRACES_COMPRESSION`
+        ///   - `OTEL_EXPORTER_OTLP_METRICS_COMPRESSION`
+        ///   - `OTEL_EXPORTER_OTLP_LOGS_COMPRESSION`
+        /// - Default value: `.none`
+        /// - Supported values: `none`, `gzip`
+        /// - Notes: Signal-specific configuration takes precedence over the general configuration.
         public var compression: Compression
+
+        /// Maximum time the exporter waits for each batch export.
+        ///
+        /// Specifies the maximum duration to wait for export operations to complete.
+        /// If an export operation takes longer than this timeout, it will be cancelled.
+        ///
+        /// - Environment variable(s):
+        ///   - `OTEL_EXPORTER_OTLP_TIMEOUT`
+        ///   - `OTEL_EXPORTER_OTLP_TRACES_TIMEOUT`
+        ///   - `OTEL_EXPORTER_OTLP_METRICS_TIMEOUT`
+        ///   - `OTEL_EXPORTER_OTLP_LOGS_TIMEOUT`
+        /// - Default value: 10 seconds.
+        /// - Notes: Signal-specific configuration takes precedence over the general configuration.
         public var timeout: Duration
+
+        /// The transport protocol for OTLP communication.
```

Comment:
What do you think about calling out the required package traits here?

Reply:
Yeah, good point on the maintenance overhead this would imply. I'm happy to leave it up to the availability attribute on the symbols.